virtual-list.js

```diff
@@ -396,6 +412,27 @@ export const RepeatsAndScrolls = Superclass => class extends Repeats
     this._container.dispatchEvent(
         new RangeChangeEvent('rangechange', {first, last}));
   }
+  /**
+   * @private
+   */
+  _containerSizeChanged(size) {
+    // Include also padding.
```

Comment:
I had to go look at the ResizeObserve spec to understand this comment. Maybe include more detail, like "We want to include the padding; ResizeObserve gives that to us in the left and top properties."

I am also surprised you are doing left + right instead of left + width?

Reply:
good point, will expand the comment. Weirdly enough left + width omits the padding-right, and right omits the padding-left O_o

Reply:
Per spec I don't actually see any way to figure out an element's padding right :(. /cc @atotic.

The spec says left = padding left, width = content width. That means automatically right = padding left + content width. So left + right = (2 * padding left) + content width. So I think this code is broken if you have uneven left/right paddings. (Or uneven top/bottom paddings.)

Maybe you just need to query the padding directly, instead of using the resize observer info? Hopefully that doesn't cause a relayout, since you'll be delivered all this stuff at a time where it's already calculated anyway?

(Although, I'm not sure why we need the padding anyway, exactly...)

Reply:
You're right, we should care of the width only

Reply:
I got initially tricked by the fact that, since we position children with `position: absolute` and set their `width: 100%`, they would get the total width of their container (padding included), so I was including that in here too... the right thing to do here is read only width/height, and if the user sets padding on the container, they'll have to include that in the children too, e.g.
```css
virtual-list {
  padding: 10px;
}
virtual-list > * {
  width: calc(100% - 20px);
}

Reply:
Hmm, that's kind of annoying, maybe worth writing a separate issue to see if we can automatically make padding on the virtual-list work.

Reply:
done https://github.com/valdrinkoshi/virtual-list/issues/77

Reply:
You are correct. There is no way to get the padding from ResizeObservation. ResizeObserver wanted to support all the CSS rects initially, but could not observe them [performantly](https://github.com/WICG/ResizeObserver/issues/4), so we ended up with content rect size & position.

Workaround for observing padding of container:
    <div id="container">
        <div id="paddingObserver" style="position:absolute; top:0;left:0;right:0;bottom:0;visibility:hidden">
RO.observe(paddingObserver);

Another note:
You are calling "scheduleUpdateView" from ResizeObserver callbacks. This means you are skipping a frame because event loop ordering is:

rAF
ResizeObserver
Paint

rAF scheduled inside ResizeObserver will execute in the next event loop, after paint. If you wanted to schedule a callback before paint, Promise microtask is the way.

If you have any further question on RO use, feel free to ping me, some stuff is just non-obvious.

Reply:
Moved the discussion in the relative issues #77 and #78 👌